_api_app/app/Sites/MultisitesMenuRenderService.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace App\Sites;
+
+use App\Shared\Helpers;
+use App\Shared\Storage;
+use App\Sites\SitesDataService;
+
+class MultisitesMenuRenderService
+{
+    private $currentSite;
+    private $isEditMode;
+
+    /**
+     * Construct SitesRenderService instance
+     *
+     * @param string $currentSite
+     * @param bool $isEditMode
+     */
+
+    public function __construct(
+        $currentSite,
+        $isEditMode
+    ){
+        $this->currentSite = $currentSite;
+        $this->isEditMode = $isEditMode;
+    }
+
+    public function getViewData()
+    {
+        $sitesDataService = new SitesDataService();
+        $sites = $sitesDataService->get();
+        $data['sites'] = [];
+        $i = 0;
+
+        foreach ($sites as $site) {
+            $isPublished = $this->isEditMode == true ? true : $site['@attributes']['published'] == 1 ? true : false;
+
+            if($isPublished) {
+                $className = $this->currentSite === $site['name'] ? 'class= "selected"' : null;
```

Comment:
Store only class name in $className variable. And output it in template if it's set.